core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java

```diff
@@ -285,6 +351,13 @@ UpgradeV4Response checkForClusterUpgradeByName(@PathParam("workspaceId") Long wo
     FlowIdentifier upgradeClusterByName(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name, String imageId,
             @AccountId @QueryParam("accountId") String accountId);
 
+    @POST
+    @Path("internal/{name}/cluster_upgrade")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = STACK_UPGRADE_INTERNAL, nickname = "upgradeClusterByNameInternal")
+    FlowIdentifier upgradeClusterByNameInternal(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name, String imageId,
```

Comment:
Can't we skip the whole workspace ID from these new API's?

Reply:
Or use accountId instead?

Reply:
I'd like to, but we have some precheck code for this and fails if there is no workspaceId. We can create a followup task for this but I didn't want to deal with this in this PR.